docs/v2/adding-new-codegenerated-resource.md

```diff
@@ -0,0 +1,83 @@
+1. If the resource is defined in the [ARM template JSON schema](https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json), or the [auto-generated ARM template JSON schema](https://schema.management.azure.com/schemas/common/autogeneratedResources.json) it is an ARM resource.
+2. If the resource is defined in a `resource-manager` folder in the [Azure REST API specs](https://github.com/Azure/azure-rest-api-specs/tree/main/specification) repo, it is an ARM resource.
+
+If the resource is not in either of the above places and cannot be deployed via an ARM template, then it's not an ARM resource and currently cannot be code generated.
+
+## Determine a resource to add
+There are three key pieces of information required before adding a resource to the code generation configuration file, and each of them can be found in the 
+[ARM template JSON schema](https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json) or the [auto-generated ARM template JSON schema](https://schema.management.azure.com/schemas/common/autogeneratedResources.json). To get started, find the entry for the resource in one of the two templates mentioned. For example, the entry for Azure Network Security Groups looks like this:
+`{ "$ref": "https://schema.management.azure.com/schemas/2020-11-01/Microsoft.Network.json#/resourceDefinitions/networkSecurityGroups" },`
+
+**Note**: In many cases there will be multiple entry's for the same resource, each with a different `api-version`. It is _strongly_ recommended that you use the latest available `api-version`.
+
+### The `name` of the resource
+You usually know this going in. In our example above, the name of the resource is `networkSecurityGroups`.
+
+### The `group` the resource is in 
+This is usually named after the Azure service, for example `microsoft.resources` or `microsoft.documentdb`. In our example entry from above, this is `microsoft.network`.
+
+### The `api-version` of the resource
+This is usually a date, sometimes with a `-preview` suffix. In our example entry from above, this is `2020-11-01`.
+
+## Adding the resource to the code generation configuration file
+The code generation configuration file is located [here](../../hack/generator/azure-arm.yaml). To add a new resource to this file, find the `exportFilters` section of the file and scroll down until you get to a block of `exportFilters` for individual resources. 
+
+Add a new `exportFilter` of kind `include-transitive` at the end of that block, right above this section:
+```yaml
+  # Exclude everything else as we are operating on an opt-in basis at the moment:
+  - action: exclude
+    because: We don't want to generate anything else, at the moment.
+```
+
+Your export filter should look like this:
+```yaml
+- action: include-transitive
+  group: <group>
+  version: v*api<api-version with dashes removed>
+  name: <resource name, with trailing "s" removed>
```

Comment:
Will every resource name become singular just by removing the `s`? 

```suggestion
  name: <singular resource name, typically just remove the trailing "s">
```

Reply:
Good catch